context/app/static/js/components/savedLists/api.ts

```diff
@@ -0,0 +1,329 @@
+import useSWR from 'swr';
+import { fetcher } from 'js/helpers/swr/fetchers';
+import { useAppContext } from 'js/components/Contexts';
+import { SavedEntitiesList, SavedEntity } from 'js/components/savedLists/types';
+import { SAVED_ENTITIES_DEFAULT } from 'js/components/savedLists/constants';
+
+/** **************************************************
+ *                  UKV API Helpers                  *
+ * ************************************************* */
+
+/**
+ * Generates API URLs for various saved list API actions
+ * @param ukvEndpoint endpoint to use as the base for API URLs
+ * @returns an API URL generator
+ */
+const apiUrls = (ukvEndpoint: string) => ({
+  get keys(): string {
+    return `${ukvEndpoint}/user/keys`;
+  },
+  get findKeys(): string {
+    return `${ukvEndpoint}/user/find/keys`;
+  },
+  key(key: string): string {
+    return `${ukvEndpoint}/user/keys/${key}`;
+  },
+});
+
+function useUkvApiURLs() {
+  const { ukvEndpoint } = useAppContext();
+  return apiUrls(ukvEndpoint);
+}
+
+async function fetchResponse({
+  url,
+  token,
+  method,
+  body,
+}: {
+  url: string;
+  token: string;
+  method: 'PUT' | 'DELETE';
+  body?: string;
+}) {
+  const response = await fetch(url, {
+    method,
+    headers: {
+      Authorization: `Bearer ${token}`,
+      'Content-Type': 'application/json',
+    },
+    body,
+  });
+
+  if (!response.ok) {
+    console.error('UKV API failed with error: ', await response.text());
+  }
+}
```

Comment:
This is pattern used elsewhere so no fault in introducing it, but could we use the fetcher in the swr helpers?